fish-rust/src/job_group.rs

```diff
@@ -100,8 +101,8 @@ pub struct JobGroup {
     /// "Simple block" groups like function calls do not have a job id.
     pub job_id: Option<JobId>,
     /// The signal causing the group to cancel or `0` if none.
-    /// Not using an `Option<NonZeroI32>` to be able to atomically load/store to this field.
-    signal: AtomicI32,
+    /// Not using an `Option<Signal>` to be able to atomically load/store to this field.
+    signal: AtomicU8,
```

Comment:
Why change from I32 to U8? I prefer I32 because signals have type int in the system interface.

Reply:
Just to match the Signal inner type. I originally wrapped Signal around NonZeroI32 but went with code purity instead since signal numbers just don't go that high even though they're returned as int by the system (largely because that's the smallest type the system really uses, especially since char isn't defined as being signed or unsigned). But I can change everything back to i32/NonZeroI32 to get rid of any downsizing safety checks.